internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -727,8 +746,62 @@ func resourceOrchestratedVirtualMachineScaleSetCreate(d *pluginsdk.ResourceData,
 	}
 
 	if v, ok := d.GetOk("encryption_at_host_enabled"); ok {
-		virtualMachineProfile.SecurityProfile = &virtualmachinescalesets.SecurityProfile{
-			EncryptionAtHost: pointer.To(v.(bool)),
+		if v.(bool) {
+			if virtualmachinescalesets.SecurityEncryptionTypesDiskWithVMGuestState == virtualmachinescalesets.SecurityEncryptionTypes(securityEncryptionType) {
+				return errors.New("`encryption_at_host_enabled` cannot be set to `true` when `os_disk.0.security_encryption_type` is set to `DiskWithVMGuestState`")
+			}
+		}
+
+		if virtualMachineProfile.SecurityProfile == nil {
+			virtualMachineProfile.SecurityProfile = &virtualmachinescalesets.SecurityProfile{}
+		}
+
+		virtualMachineProfile.SecurityProfile.EncryptionAtHost = pointer.To(v.(bool))
+	}
+
+	secureBootEnabled := d.Get("secure_boot_enabled").(bool)
+	vtpmEnabled := d.Get("vtpm_enabled").(bool)
+
+	if securityEncryptionType != "" {
```

Comment:
This if-condition seems complicated, I'm thinking that if we could rewrite it like

1. Validate the constraints, then early return.
2. If secureBootEnabled
3. If vTPM enabled

Reply:
I think it may make sense. However, this code is totally leveraged from [linux_virtual_machine_scale_set_resource.go](https://github.com/hashicorp/terraform-provider-azurerm/blob/a71bbd5bd4b923369d5e69d4136f3b9bd2c8c57a/internal/services/compute/linux_virtual_machine_scale_set_resource.go#L327). As the code has already worked properly for a long time, I suggest using the same code in this PR. WDYT?